content/docs/guides/neon-auth.md

```diff
@@ -0,0 +1,180 @@
+  const { userId } = event;
+
+  // Fetch user data from auth provider
+  const auth = new AuthProvider();
+  const user = await auth.getUser(userId);
+
+  // Update database
+  await db.query(
+    `
+    INSERT INTO users (id, email, name)
+    VALUES ($1, $2, $3)
+    ON CONFLICT (id) DO UPDATE 
+    SET email = $2, name = $3
+  `,
+    [user.id, user.email, user.name]
+  );
+});
+```
+
+### After Neon Auth
+
+With Neon Auth, much of this complexity is eliminated. Since user data is automatically synced to `neon_auth.users_sync`, you can just create the todo:
+
+```typescript
+async function createTodo(userId: string, task: string) {
+  return db.query(
+    `INSERT INTO todos (task, user_id) 
+     VALUES ($1, $2)`,
+    [task, userId]
+  );
+}
+```
+
+## Getting started
+
+1. From the Neon Console, navigate to the **Auth** tab. Choose your provider and click **Connect**. Currently, only Stack Auth is available for Early Access users.
+1. You'll be asked to authenticate and select the project you want to integrate with.
+1. Once connected, you'll see the integration view. This shows your synced users, connection status, and quick links to your provider's documentation and console to help configure your application (e.g. SSO or API keys)
+
+Here's an example of Neon Auth with Stack Auth. As we add more providers and features, this interface will continue to evolve.
+
+![Neon Auth with stackauth deployed](/docs/guides/identity_stackauth.png)
+
+## Best practices
+
+### Foreign keys and the users_sync table
+
+Since the `neon_auth.users_sync` table is updated asynchronously, there may be a brief delay (usually less than 1 second) before a user's data appears in the table. Consider this possible delay when deciding whether to use foreign keys in your schema.
+
+If you do choose to use foreign keys, make sure to specify an `ON DELETE` behavior that matches your needs: for example, `CASCADE` for personal data like todos or user preferences, and `SET NULL` for content like blog posts or comments that should persist after user deletion.
+
+```sql
+-- For personal data that should be removed with the user (e.g., todos)
+CREATE TABLE todos (
+    id SERIAL PRIMARY KEY,
+    task TEXT NOT NULL,
+    user_id UUID NOT NULL REFERENCES neon_auth.users_sync(id) ON DELETE CASCADE,
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+);
+
+-- For content that should persist after user deletion (e.g., blog posts)
+CREATE TABLE posts (
+    id SERIAL PRIMARY KEY,
+    title TEXT NOT NULL,
+    content TEXT NOT NULL,
+    author_id UUID REFERENCES neon_auth.users_sync(id) ON DELETE SET NULL,
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+);
+```
+
+### Querying user data
+
+When querying data that relates to users:
+
+- Use LEFT JOINs instead of INNER JOINs with the `users_sync` table in case of any sync delays. This ensures that all records from the main table (.e.g. posts) are returned even if there's no matching user in the `users_sync` table yet.
```

Comment:
```suggestion
- Use LEFT JOINs instead of INNER JOINs with the `users_sync` table in case of any sync delays. This ensures that all records from the main table (e.g., posts) are returned even if there's no matching user in the `users_sync` table yet.
```